src/site/antora/modules/ROOT/partials/log4j-features.adoc

```diff
@@ -0,0 +1,47 @@
+////
+    Licensed to the Apache Software Foundation (ASF) under one or more
+    contributor license agreements.  See the NOTICE file distributed with
+    this work for additional information regarding copyright ownership.
+    The ASF licenses this file to You under the Apache License, Version 2.0
+    (the "License"); you may not use this file except in compliance with
+    the License.  You may obtain a copy of the License at
+
+         http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+////
+
+Batteries included::
+Log4j bundles a rich set of components to assist various use cases.
+* xref:manual/appenders.adoc[Appenders] targeting files, network sockets, databases, SMTP servers, etc.
+* xref:manual/layouts.adoc[Layouts] that can render CSV, HTML, JSON, Syslog, etc. formatted outputs
+* xref:manual/filters.adoc[Filters] based on log event rates, regular expressions, scripts, time, etc.
+* xref:manual/lookups.adoc[Lookups] for accessing system properties, environment variables, log event fields, etc.
+
+API separation::
+The API for Log4j (i.e., `log4j-api`) is separate from the implementation (i.e., `log4j-core`), making it straightforward for application developers which classes and methods to use while ensuring forward compatibility.
```

Comment:
Something is wrong in "making it straightforward for application developers which classes and methods to use while ensuring forward compatibility". Perhaps "straightforward to use in a correct and future-proof (forward-compatible) way"?

Reply:
I think this whole paragraph doesn't make sense at all on the benefits of API-vs-implementation separation. I will rewrite this.

Reply:
Yes, I guess we should start with the advantages of Log4j API, followed by the advantages of Log4j Core. So in order:

1. "No vendor lock-in": the Log4j API and its reference implementation are separate and Log4j API is in permanent LTS-mode. People can use a stable API without breaking changes, while at the same time profiting of major rewrites of the reference implementation.
2. "A feature-rich API": the API provides 100+ **different** logging methods, even if we ignore the vararg expansions. These methods can even be repurposed to obtain different semantics, using the `Message` and `MessageFactory` abstractions.
3. "Batteries included", "Reliability" and so on.

Reply:
> I think this whole paragraph doesn't make sense at all on the benefits of API-vs-implementation separation. I will rewrite this.

I believe it can make sense, but you're right that it's at best unclear. I would suggest something closer to:

> In addition to the Log4j Core reference implementation, Log4j provides a stable facade (i.e., `log4j-api`) which is separate from that implementation, making it straightforward for developers to use Log4j while being confident on a high degree of API forward-compatibility.